overlord/snapstate/snapmgr.go

```diff
@@ -319,7 +319,10 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 	pb := &TaskProgressAdapter{task: t}
 	err = m.backend.RemoveSnapFiles(ss.placeInfo(), pb)
 	if err != nil {
-		return err
+		st.Lock()
+		t.Errorf("cannot remove snap file: %s, retrying", err)
```

Comment:
That might potentially end up a bit confusing depending on what is inside err, and it is also slightly misguiding in that it points out a retrying is in progress while it may take quite a while. I suggest:

`"cannot remove snap file, will retry: %v"`